packages/block-editor/src/components/line-height-control/style.scss

```diff
@@ -5,4 +5,14 @@
 		display: block;
 		max-width: 60px;
 	}
+
+	// TODO: This may be easier if we can replace TextControl with NumberControl and use prop `size="__unstable-large"`
```

Comment:
I think you make a good point here. The fact that this control is also not a `NumberControl` means you can't drag up/down like most other number related inputs. This catches me out a lot and has also come up in a few team chats and reviews of the typography panel previously.

Quickly switching out the `TextControl` for the `NumberControl` appeared to work fine for me.

Reply:
Oh that's good to know! I'll want to discuss with @ciampo the implications of using an experimental component (`NumberControl`) inside a stable component though, I already got caught in that trap once 😆 

I added it to #36230 to keep track of the issue.

Reply:
I'd be definitely up for switching to using `NumberControl` here — not only if makes sense in terms on semantics and UX, but the code would also look cleaner.

Regarding using experimental components inside stable components, I'm generally ok with that, especially if the experimental component remains as an implementation detail of the stable component, that can be swapped/changed without introducing breaking changes to the public APIs of the stable component.

Alternatively, I'd try to see if we can apply the "size" normalisation to the `TextControl` component in the `@wordpress/components` package, rather than doing in in the block editor